mwdb/model/karton.py

```diff
@@ -85,6 +86,27 @@ def processing_in(self):
     def get(cls, analysis_id):
         return db.session.query(KartonAnalysis).filter(KartonAnalysis.id == analysis_id)
 
+    @classmethod
+    def get_or_create(cls, analysis_id, initial_object):
+        analysis = cls.get(analysis_id).first()
+
+        if analysis is not None:
+            return analysis, False
+
+        db.session.begin_nested()
```

Comment:
maybe move this two lines lower (or even inside `try:`)? Just in case there is an unexpected exception (for example, KartonAnalysis ctor throws). I know it's unlikely but just to keep the "unguarded" window small.

Or is it necessary to create KartonAnalysis() after session begin()?

Reply:
Actually that try..catch block is meant only to catch the IntegrityErrors when constraints are not fulfilled because of concurrent transaction. That's also why we are starting nested transaction to recover from that situation.

If something goes wrong during construction of KartonAnalysis object, that situation is unexpected and should result in interruption of further request processing, unhandled exception and ISE 500 sent to the end user.